### PR TITLE
chore: improvements for header actions on artwork screen

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [] 

### Demo
| Before | After |
| ------------- | ------------- |
| before-1 | after-1 | 
| before-2 | after-2 |

### Video
https://user-images.githubusercontent.com/3513494/206737595-6b6598d3-0207-46ba-8fb2-91fa5bc056ba.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Improvements for header actions on artwork screen - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/ArtworkScreenHeader.tsx [hunk 1]
```diff
@@ -117,19 +104,16 @@ const ArtworkScreenHeader: React.FC<ArtworkScreenHeaderProps> = ({ artwork }) =>
...
 
       <Flex flexDirection="row" alignItems="center">
-        <Button
-          icon={<SaveIcon isSaved={!!isSaved} />}
-          size="small"
-          variant="text"
+        <Touchable
           accessibilityRole="button"
-          accessibilityLabel={saveButtonText()}
+          accessibilityLabel="Save button"
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/ArtworkScreenHeader.tsx:109]: super minor but can we change the a11y label to:
```suggestion
          accessibilityLabel="Save artwork"
```

**Review by gkartalis — APPROVED** [review]
LGTM!

File: src/app/Scenes/Artwork/Components/ArtworkScreenHeader.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/app/Scenes/Artwork/Components/ArtworkScreenHeader.tsx]: Done

**Review by gkartalis — APPROVED** [review]
🛳️

**Review by dblandin — APPROVED** [review]
Nice! Looks great 💯

**Pull request merged**